bundles/org.openhab.binding.nanoleaf/src/main/java/org/openhab/binding/nanoleaf/internal/model/Layout.java

```diff
@@ -41,12 +47,74 @@ public void setSideLength(int sideLength) {
         this.sideLength = sideLength;
     }
 
-    public List<PositionDatum> getPositionData() {
+    public @Nullable List<PositionDatum> getPositionData() {
         return positionData;
     }
 
     public void setPositionData(List<PositionDatum> positionData) {
         this.positionData = positionData;
     }
 
+    /**
+     * Returns an text representation for a canvas layout.
+     *
+     * Note only canvas supported currently due to its easy geometry
+     * @return a String containing the layout
+     */
+    public String getLayoutView() {
+        if (positionData!=null) {
+            String view = "";
+
+            int minx = Integer.MAX_VALUE, maxx = Integer.MIN_VALUE, miny = Integer.MAX_VALUE, maxy = Integer.MIN_VALUE;
+
+
+            for (int index = 0; index < numPanels; index++) {
+                if (positionData != null) {
+                    @Nullable PositionDatum panel = positionData.get(index);
+
+                    if (panel != null) {
+                        if (panel.getPosX() < minx) minx = panel.getPosX();
```

Comment:
Please use braces, I suppose the formatter will break the elegance of current layout anyway ;-)